assets/cluster-monitoring-operator/prometheus-rule.yaml

```diff
@@ -28,6 +28,37 @@ spec:
       for: 15m
       labels:
         severity: warning
+    - alert: HighlyAvailableWorkloadIncorrectlySpread
+      annotations:
+        description: Workload {{ $labels.namespace }}/{{ $labels.workload }} is incorrectly
+          spread across mutliple nodes which breaks high-availability requirements.
+          There are {{ $value }} pods on node {{ $labels.node }}, where there should
+          only be one. Since the workload is using persistent volumesm it is not fixable
+          by operators, but you can follow the guidelines provided in the runbook
+          of this alert to fix this issue.
+        runbook_url: https://github.com/openshift/runbooks/blob/master/alerts/HighlyAvailableWorkloadIncorrectlySpread.md
```

Comment:
Holding until this runbook is created and linked to the Bugzilla.

/hold

Reply:
I believe this is one of the first cases where we link to a runbook. We need to document this and probably add it to KB. Otherwise, this might not be used at all.

Reply:
Yes, this is the first alert for which we want to add a runbook downstream. I am fine with documenting the process, but one thing I want to make sure of is that we want to move forward with leveraging the effort from OSD folks in https://github.com/openshift/runbooks and make it our central place for runbooks. We might even be able to automate the inclusion of runbooks from this repository in the future if we were to go that way.

That said, I have two more concerns with that, first, we should make it clear that the runbooks in this repository will also be available to external customers, and last from a UX perspective, do we really want to redirect our customers to Github pages instead of having the runbooks integrated to the console?

I believe since it is quite urgent to get this alert in, we can start by adding the runbook to the downstream repo and improve the process in the future.

Reply:
Personally, I wouldn't want to have runbooks integrated into the Console. What if Console is down or a route is inaccessible? Usually, for this reason, runbooks are hosted externally to infrastructure running services. 

I have no idea how this will be solved in OpenShift. Will OpenShift use Knowledge Base for it, or GitHub repository, or integrate it with Console? It is an initiative that needs to be done on a higher level than just the monitoring team and with consultation from Architects as well as SREP. I think writing OEP might be a good start.

For now, as we don't have an established process in place, I wouldn't say that lack of runbook should be blocking anything. But without any runbook we shouldn't have a link to it in alert annotations too. I would propose to remove runbook link from this PR and add another one with only a runbook link and some sort of check which we could run in CI to validate if the link is leading to anything. If there is no hurry, adding this check could be done as part of this PR.

Reply:
Good point :+1:, I will ask Christian if there is any plan with regard to that today.

I would personally prefer to add the runbook to this PR since it wouldn't make sense to have this particular alert without a runbook. That said, I can already open a follow-up PR to add the validation process and we can get it merged before this one.